docs/core/compatibility/core-libraries/7.0/max-precision-numeric-format-strings.md

```diff
@@ -0,0 +1,113 @@
+description: Learn about the .NET 7 breaking change in core .NET libraries where the maximum precision for numeric format strings was changed to 999,999,999.
+ms.date: 09/02/2022
+---
+# Maximum precision for numeric format strings
+
+The maximum precision when formatting numbers as strings using `ToString` and `TryFormat` has been changed from <xref:System.Int32.MaxValue?displayProperty=nameWithType> to 999,999,999. (The maximum precision was [previously changed](../6.0/numeric-format-parsing-handles-higher-precision.md) to <xref:System.Int32.MaxValue?displayProperty=nameWithType> in .NET 6.)
+
+In addition, the maximum exponent allowed when parsing an <xref:System.Int64> from a string has been limited to 999,999,999.
+
+## Previous behavior
+
+In .NET 6, the standard numeric format parsing logic was limited to a precision of <xref:System.Int32.MaxValue?displayProperty=nameWithType> or less. The intended behavior was to throw a <xref:System.FormatException> for any precision larger than <xref:System.Int32.MaxValue?displayProperty=nameWithType>. However, due to a bug, .NET 6 didn't throw that exception for some such inputs. The intended behavior was:
+
+```csharp
+double d = 123.0;
+
+d.ToString("E" + int.MaxValue.ToString()); // Doesn't throw.
+
+long intMaxPlus1 = (long)int.MaxValue + 1;
+string intMaxPlus1String = intMaxPlus1.ToString();
+Assert.Throws<FormatException>(() => d.ToString("E" + intMaxPlus1String)); // Throws.
+```
+
+In addition, there was no limit on the exponent size when parsing an <xref:System.Int64> from a string.
+
+## New behavior
+
+Starting in .NET 7, .NET supports precision up to 999,999,999. A <xref:System.FormatException> is thrown if the precision is greater than 999,999,999. This change was implemented in the parsing logic that affects all numeric types.
+
+```csharp
+double d = 123.0;
+Assert.Throws<FormatException>(() => d.ToString("E" + int.MaxValue.ToString())); // Throws.
+
+long intMaxPlus1 = (long)int.MaxValue + 1;
+string intMaxPlus1String = intMaxPlus1.ToString();
+Assert.Throws<FormatException>(() => d.ToString("E" + intMaxPlus1String)); // Throws.
+
+d.ToString("E999999999"); // Doesn't throw.
+
+d.ToString("E00000999999999"); // Doesn't throw.
+```
+
+In addition, if you attempt to parse an <xref:System.Int64> with an exponent greater than 999,999,999 from a string, a <xref:System.FormatException> is thrown.
+
+## Version introduced
+
+.NET 7 RC 1
+
+## Type of breaking change
+
+This change can affect [binary compatibility](../../categories.md#binary-compatibility).
+
+## Reason for change
+
+The behavior that was introduced in .NET 6 was intended to throw a <xref:System.FormatException> for any precision larger than <xref:System.Int32.MaxValue?displayProperty=nameWithType>. However, due to a bug, it did not throw that exception for some inputs larger than the maximum. This change fixes the bug by limiting the precision to 999,999,999.
```

Comment:
I don't like this phrase:

> This change fixes the bug by limiting the precision to 999,999,999.

The fix isn't the limit being set, the limit was in the previous version. The fix is the exception is thrown.